react-form-controls/src/registration-form.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+export default class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    console.log('state: ', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={ this.handleSubmit }>
+        <label htmlFor="username">Username:</label>
```

Comment:
the `htmlFor` prop should match the `id` prop of the associated `input`. So if you are using `htmlFor` on the label the `input` should have a matching `id`.